### ISSUE TITLE
broken water translucency on station15 map

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#120]
See the waterfall and the pond (two different shaders):

[![incorrect](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235232_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235232_000.jpg)

Note there is also a problem with depth:

[![incorrect](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_223241_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_223241_000.jpg)

[![incorrect](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_223259_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_223259_000.jpg)

[![incorrect](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235106_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235106_000.jpg)

This is a third and similarly broken translucent water shader:

[![incorrect](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235113_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug/unvanquished_2018-08-28_235113_000.jpg)

**Comment by illwieckz** [comment]
Here is one example of buggy material:

```c
// Crappy water with leaves
textures/station15_custom/water2
{
	qer_editorImage textures/station15_custom_src/leaves_d
	qer_trans .3
	q3map_globaltexture
	entityMergable
	surfaceparm nobuild
	surfaceparm trans
	surfaceparm water
	surfaceparm nonsolid
	surfaceparm nolightmap
	cull disable
	tesssize 64
	deformVertexes wave 64 sin .8 1.5 2 .8
	{
		map textures/station15_custom_src/water_d
		rgbGen const ( .8 .8 .3 )
		alphaGen const .2
		blendfunc blend
		tcMod scroll .03 .001
	}
	{
		map textures/station15_custom_src/waterfall_r
		alphagen const .3
		blendfunc blend
		rgbGen const ( .8 .9 1 )	// ← this line
		tcmod scale .2 1
		tcGen environment
	}
	{
		map textures/station15_custom_src/leaves_d
		blendfunc blend
		AlphaFunc GE128
		DepthWrite
	}
}
```

Notice the `rgbGen const ( .8 .9 1 )` line I identified.

This is what produces the original material:

[![waterbug](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232245_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232245_000.jpg)

This is what produces the same material but with that line commented out (disabled):

[![waterbug](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232342_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232342_000.jpg)

This is what produces the same material but with that line moved before the `alphagen const .3` line:

[![waterbug](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232428_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_232428_000.jpg)

I see no difference between without or before. Perhaps we have a bug where `alphaGen` deletes rgb channels and where `rgbGen` deletes alpha channel.

**Comment by illwieckz** [comment]
It looks like we have a fix in #181:

[![fixedbug](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_235500_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/maps/station15/waterbug//unvanquished_2019-03-10_235500_000.jpg)
